packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -0,0 +1,572 @@
+///                 optionsBuilder: (BuildContext context, AutocompleteOnSelected<String> onSelected, List<String> options) {
+///                   return Material(
+///                     elevation: 4.0,
+///                     child: SizedBox(
+///                       height: 200.0,
+///                       child: ListView(
+///                         padding: EdgeInsets.all(8.0),
+///                         children: options.map((String option) => GestureDetector(
+///                           onTap: () {
+///                             onSelected(option);
+///                           },
+///                           child: ListTile(
+///                             title: Text(option),
+///                           ),
+///                         )).toList(),
+///                       ),
+///                     ),
+///                   );
+///                 },
+///               ),
+///               RaisedButton(
+///                 onPressed: () {
+///                   FocusScope.of(context).requestFocus(new FocusNode());
+///                   if (!_formKey.currentState.validate()) {
+///                     return;
+///                   }
+///                   showDialog<void>(
+///                     context: context,
+///                     builder: (BuildContext context) {
+///                       return AlertDialog(
+///                         title: Text('Successfully submitted'),
+///                         content: SingleChildScrollView(
+///                           child: ListBody(
+///                             children: <Widget>[
+///                               Text('DropdownButtonFormField: "$_dropdownValue"'),
+///                               Text('TextFormField: "${_textEditingController.text}"'),
+///                               Text('AutocompleteCore: "$_autocompleteSelection"'),
+///                             ],
+///                           ),
+///                         ),
+///                         actions: <Widget>[
+///                           FlatButton(
+///                             child: Text('Ok'),
+///                             onPressed: () {
+///                               Navigator.of(context).pop();
+///                             },
+///                           ),
+///                         ],
+///                       );
+///                     },
+///                   );
+///                 },
+///                 child: Text('Submit'),
+///               ),
+///             ],
+///           ),
+///         ),
+///       ),
+///     );
+///   }
+/// }
+/// ```
+/// {@end-tool}
+class AutocompleteCore<T> extends StatefulWidget {
+  /// Create an instance of AutocompleteCore.
+  ///
+  /// [fieldBuilder] and [optionsBuilder] must not be null.
+  const AutocompleteCore({
+    required this.fieldBuilder,
+    required this.optionsBuilder,
+    required this.buildOptions,
+    AutocompleteOptionToString<T>? displayStringForOption,
```

Comment:
I was getting "the default value of an optional parameter must be constant" when I tried that.  I think it's because it can't know the <T> type at that point...  I'm going to change the _defaultStringForOption method to take a `dynamic`, then it seems to work.